api/config/v1alpha1/shared_types.go

```diff
@@ -77,6 +77,12 @@ type KubernetesPodSpec struct {
 	// +optional
 	Annotations map[string]string `json:"annotations,omitempty"`
 
+	// Labels are the labels that should be tagged to the pods.
+	// By default, no pod labels are tagged.
+	//
```

Comment:
made above comments since we do have default labels https://github.com/envoyproxy/gateway/blob/773bdfbefc769f51779d5056547e88d12e1d9508/internal/infrastructure/kubernetes/proxy/resource.go#L62, we dont want to the user to rm these